### DISCUSSION TITLE
Question about useReducerAtom and derived atoms

### DISCUSSION THREAD

**Discussion by mo-lon** [pmndrs/jotai#2672] [Q&A] [answered]
Hi All - i'm starting out with Jotai and had a question around re-renders and usage of useReducerAtom in this contrived example

https://codesandbox.io/p/sandbox/inspiring-haze-37n98f

I would have expected the components to only re-render when their state changed but doesnt look like thats the case. The atoms are derived from state slices and the overall state is updated by the dispatcher

**Comment by rothsandro** [comment, marked as answer]
Your `useReducerAtom` hook reads the whole atom state, that's why the component re-render. Replace it with `useSetAtom` which doesn't return the atom value:

```diff
- const [, setState] = useAtom(anAtom);
+ const setState = useSetAtom(anAtom);
```

**Reply by mo-lon** [reply]
Thanks, that did the trick
